### DISCUSSION TITLE
Thermal expansion

### DISCUSSION THREAD

**Discussion by amosaha** [idaholab/moose#19314] [Q&A General] [answered]
hello,
I want to couple a problem of heat transfer and thermal expansion. My geometric model is a cylinder. I want to fix the two bottom surfaces of the cylinder.Observe the deformation of the middle section. How should I set the BCS? My code is as follows, fix 1 and fix 2 are the bottom surfaces of the cylinder that I expected to be fixed.
[code.txt](https://github.com/idaholab/moose/files/7495758/code.txt)

![02A687E2139F9DF7D772FEED5A3C5CA0](https://user-images.githubusercontent.com/93416273/140717944-7de9c9be-d2d7-4dfc-a7d7-3757fc3bbda8.png)
![image](https://user-images.githubusercontent.com/93416273/140718682-c90a9eb7-6c94-4810-93ee-a0803fe76c23.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

You ll find a lot of examples of fixing boundaries in space in the tensor_mechanics module. 
I think you just need to set a dirichlet BC on the relevant displacement variable. 

Guillaume

**Reply by amosaha** [reply]
I want to fix two bottom surfaces, but most of the models are two-dimensional problems. Where can I find relevant examples for reference? My program does not converge after running. I think it may be related to BCS.Thanks.

**Reply by GiudGiud** [reply]
it should work the same in 3D. What is the normal for these two surfaces?

**Reply by amosaha** [reply]
(0，0，1)and(0，0，-1). I just limited the z-coordinates of the two faces, and I don't know how to control the entire face.So I didn’t use the normal vector.

**Reply by GiudGiud** [reply]
aren't they in the same plane? You only want to constrain the bottom right?
regardless a DirichetBC on disp_z should be enough.

**Reply by amosaha** [reply]
I want to limit the upper and lower bottom surfaces of the cylinders, but only restricting their z coordinate system is not enough.

**Reply by GiudGiud** [reply]
you want to restrict them in X and Y too?
That just means adding DirichletBC on disp_x and disp_y
